File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int start1 = 0;
        int start2 = 0; 
        string answer = ""; 
        while(start1 < word1.length() && start2 < word2.length()){
            answer += word1[start1++]; 
            answer += word2[start2++]; 
        }

        if(word1.length() > word2.length()){
            answer += word1.substr(start1); 
        } else{
            answer += word2.substr(start2); 
        }

        return answer; 
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder(); 
        int p1 = 0, p2 = 0; 
        int n = word1.length(), m = word2.length(); 

        while(p1 < n && p2 < m){
            char a = word1.charAt(p1); 
            char b = word2.charAt(p2); 


            sb.append(a).append(b); 
            p1++;
            p2++; 
        }

        if(p1 < n) sb.append(word1.substring(p1));
        if(p2 < m) sb.append(word2.substring(p2)); 

        return sb.toString(); 
    }
}